crates/ruff_server/src/server.rs

```diff
@@ -47,14 +49,31 @@ impl Server {
         let init_params: types::InitializeParams = serde_json::from_value(params)?;
 
         let client_capabilities = init_params.capabilities;
-        let server_capabilities = Self::server_capabilities(&client_capabilities);
+        let position_encoding = Self::find_best_position_encoding(&client_capabilities);
+        let server_capabilities = Self::server_capabilities(position_encoding);
+
+        let AllSettings {
+            global_settings,
+            mut workspace_settings,
+        } = AllSettings::from_value(init_params.initialization_options.unwrap_or_default());
+
+        let mut workspace_for_uri = |uri| {
+            let settings = workspace_settings.remove(&uri).unwrap_or_else(|| {
+                tracing::warn!("No workspace settings found for {uri}");
```

Comment:
I don't think this should be a warning, it should be at info or debug. Neovim doesn't have the concept of workspaces builtin so it will always default to providing it as global settings.